h5py/h5d.pyx

```diff
@@ -151,6 +151,55 @@ def open(ObjectID loc not None, char* name, PropID dapl=None):
     """
     return DatasetID(H5Dopen(loc.id, name, pdefault(dapl)))
 
+IF HDF5_VERSION >= (1, 14, 0):
+    @with_phil
+    def read_multi(count, list dataset_ids, list mspace_ids, list fspace_ids,
+        list type_ids, list bufs not None, PropID dxpl=None):
```

Comment:
These parameters are lists instead of numpy arrays to avoid an issue with the destination buffers. 

Specifically, the individual destination buffer for each dset read should be a numpy array. If `bufs` were itself also a numpy array containing these arrays, then it would copy those arrays instead of pointing to the same memory, and the original destination arrays the user created wouldn't get populated.

Reply:
Thanks, that makes sense.

The Python function probably shouldn't take `count` as the first parameter - it's necessary in C to tell you how many items you can read after a pointer, but lists have a length.

But if that's the case, I'd probably do a check up front that the lists are all the same length, and fail with a clear error if not.